### PR TITLE
Avoid running coverage job if rs files weren't changed 

### PR DESCRIPTION
**Pull request by VitaliyaIoffe**
## Description

This patch allows to avoid run of coverage job if rust files weren't change
For avoiding duplication between workflows I have placed the job separately, which is aimed to provide info about any_changed rust files.

**I'd like to test it more, so, now it is only draft.** 
Examples: 
https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2168406643
https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2168401487
Close pyrsia/pyrsia#577

# PR Checklist

- [ ] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [ ] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [ ] I've included a good title and brief description along with how to review them.
- [ ] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [ ] I've requested a review  from `pyrsia/collaborators`.

## Code Contributions

- [ ] I've built the code `cargo build --all-targets` successfully.
- [ ] I've run the unit tests `cargo test --workspace` and everything passes.

### CODE REVIEW

File: .github/workflows/any_changed_rust_files.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  workflow_call:
+    outputs:
+      any_changed:
+        description: "Any changed rust files"
+        value: ${{ jobs.rust-files.outputs.any_changed }}
+
+jobs:
+  rust-files:
+      runs-on: ubuntu-latest
```
Comment (sbtaylor15) [inline .github/workflows/any_changed_rust_files.yml:12]: @VitaliyaIoffe is the runs-on needed?  If that could be removed then this works for both linux and osx making it 100% generic.

Comment (VitaliyaIoffe) [reply .github/workflows/any_changed_rust_files.yml:12]: Why do we need this workflow for osx? You can call this job from another osx workflow to check that there are changed rust files. 
I think we can provide os type by input value (and it is possible to use default value). 
I'm not sure but as I remember, also, we can do something like this
`runs-on: [runner1, runner2, runner3]`

Comment (prince-chrismc) [reply .github/workflows/any_changed_rust_files.yml:12]: To run multiple os... https://github.community/t/create-matrix-with-multiple-os-and-env-for-each-one/16895/2 :wink:

File: .github/workflows/any_changed_rust_files.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Changed rust files
+
+on:
+  workflow_call:
```
Comment (prince-chrismc) [inline .github/workflows/any_changed_rust_files.yml:4]: Just out of curiosity did you consider a local composite action instead? Usually if it's only one job that's more efficient

https://dev.to/jameswallis/using-github-composite-actions-to-make-your-workflows-smaller-and-more-reusable-476l

File: .github/workflows/any_changed_rust_files.yml (hunk 1 above)
Comment (sbtaylor15) [reply .github/workflows/any_changed_rust_files.yml:12]: @VitaliyaIoffe just thinking ahead on being able to re-use this `any_changed_rust_files.yaml` in other workflows.  No specific use case in mind at this time.   Leaving it as-is for this PR works for me.

**Review by sbtaylor15 — APPROVED** [review]
changes, as they stand, look good.  Minor requested changes can be done later if needed.

**Comment by VitaliyaIoffe** [comment]
> changes, as they stand, look good. Minor requested changes can be done later if needed.

Sorry, it was a busy week. I haven't answered the @prince-chrismc question yet.

File: .github/workflows/any_changed_rust_files.yml (hunk 2 above)
Comment (VitaliyaIoffe) [reply .github/workflows/any_changed_rust_files.yml:4]: I haven't seen any articles about how many resources are needed for composite vs reusable. Maybe a simple composite action is really better for this change.

**Comment by prince-chrismc** [comment]
I'd prefer to see this as a Composite Action since that is better, but if you don't have the time we can merge this in for now and improve on it later 👍

**Comment by VitaliyaIoffe** [comment]
> I'd prefer to see this as a Composite Action since that is better, but if you don't have the time we can merge this in for now and improve on it later 👍

I am going to change reusable wf to composite under this pr :)

**Pull request closed without merging**